Frontend refactore productspages and minimum requirement.

Psypres opened the pull request.

Psypres committed:
refactor pages for scalability and maintain.

- added light/dark mode.
- minor styling, for better debugging UI.
- added toast.

Psypres committed:
styling amd better handling.

Psypres committed:
last few need to haves

MSFHdesign commented:
there is a issue when you on / click on the product page then move to edit, it shows you nothing this needs to be fixed before it's production ready.

Psypres committed:
fixed the issue with edit not working

now, @Michael i did the fix but refactoring for stronger TS in Product form and added the data in the product.value :)

MSFHdesign commented:
@MSFHdesign im proud of you! GJ!
